sshd.c

```diff
@@ -776,15 +776,28 @@ privsep_preauth(Authctxt *authctxt)
 		    posix_spawn_file_actions_adddup2(&actions, tmp_sock, STDOUT_FILENO) != 0 ||
 		    posix_spawn_file_actions_adddup2(&actions, pmonitor->m_recvfd, PRIVSEP_MONITOR_FD) != 0 ||
 		    posix_spawn_file_actions_adddup2(&actions, pmonitor->m_log_sendfd, PRIVSEP_LOG_FD) != 0 )
-			error("posix_spawn initialization failed");		
-		else {
+			fatal("posix_spawn initialization failed");		
+		
+		{
 			char** argv = privsep_child_cmdline(0);
 			if (__posix_spawn_asuser(&pid, argv[0], &actions, NULL, argv, NULL, SSH_PRIVSEP_USER) != 0)
-				error("%s, posix_spawn failed", __func__);
-			else
-				debug2("Network child is on pid %ld", (long)pid);
+				fatal("%s, fork of unprivileged child failed", __func__);
+			
 			posix_spawn_file_actions_destroy(&actions);
 		}
+
+		debug2("Network child is on pid %ld", (long)pid);
+
+		pmonitor->m_pid = pid;
+		if (have_agent) {
+			r = ssh_get_authentication_socket(&auth_sock);
+			if (r != 0) {
+				error("Could not get agent socket: %s",
```

Comment:
Since agent is optional and we proceed with connection, the log should be warning instead of error?

Reply:
The logic is triggered only when agent is configured in sshd_config. This code exists in the Unix's fork version and we missed to include it before..